### ISSUE TITLE
How to enforce entering debug mode on reset

### ISSUE THREAD

**Issue by mwachs5** [riscv/riscv-debug-spec#242]
Example: Debugger wants to trigger on some bad scenario, so it places  a breakpoint on the "bad scenario handler". Let's say bad scenario generally happens shortly after reset. Then core gets reset (by someone pressing reset button, or SW reset... essentially something out of debugger's control). Debugger will notice the hart got reset because of `hasreset`, but as the spec is currently written the HWBP set up to trigger on bad scenario could be lost and run past before debugger notices.

**Comment by timsifive** [comment]
Simplest solution:
Add a reset-proof halt-on-reset bit to DCSR in each hart. When 1, the hart will enter debug mode immediately out of reset.

**Comment by timsifive** [comment]
I'd like to get this feature in before we ratify.

I think either bit 27 in dmcontrol, or bit 3 in DCSR. Preference? It feels like it belongs more in dmcontrol so it's easier to think of it as not being reset when the hart is.

**Comment by mwachs5** [comment]
If it's in `dmcontrol`, it's global. What does that mean then w.r.t. each hart?

**Comment by timsifive** [comment]
Good point. If it's in dmcontrol, there would have to be set-halt-on-reset
and clear-halt-on-reset bits. DCSR seems more clean from that point of
view. Is there any issue with saying this one bit persists across reset?

**Comment by timsifive** [comment]
That makes me think the bits have to be in DM then, so some way of
accessing them will probably be dmcontrol. Do you agree?
